packages/expo-notifications/ios/EXNotifications/Notifications/EXNotificationCenterDelegate.m

```diff
@@ -191,7 +191,10 @@ - (void)addDelegate:(id<EXNotificationsDelegate>)delegate
         // completion handler doesn't need to do anything
       }];
     }
-    [_pendingNotificationResponses removeAllObjects];
+    // Remove EXUserNotificationManager check when LegacyNotifications are no longer supported
+    if (![NSStringFromClass([delegate class]) isEqual:@"EXUserNotificationManager"]) {
+      [_pendingNotificationResponses removeAllObjects];
```

Comment:
Does this mean that for anyone using LegacyNotifications, `_pendingNotificationResponses` would pile up on and on?

Reply:
That is true, but to some degree this pull request does not affect this behavior — it's already like this on `master`. `addDelegate:` which I'm modifying here is only called once, at the beginning of the lifetime of the app, so the _piling up_ is more caused by https://github.com/expo/expo/pull/9478/files#diff-cf2b449269635708f84869f2a2bd94d84f6cedeaec00427e2f0c7687c814ace7 than this.

One possible solution I'd have to this problem (if we decided that is a problem) would be remembering not all notification responses but only the last one. This would give us the advantage of not piling responses up in exchange for _information incompleteness_ (which in fact may be a good thing — if I'm not mistaken, right now, if `expo-notifications` listener is added some time after the app is running, after the user acts on some notifications, the new listener will get all these responses — which may be not what we want, why would a late listener have to receive old responses).

Reply:
> the piling up is more caused by https://github.com/expo/expo/pull/9478/files#diff-cf2b449269635708f84869f2a2bd94d84f6cedeaec00427e2f0c7687c814ace7 than this

that is true 😅 my bad

I like that proposed solution, i don't see why someone might care about any response except the most recent, but that's out of scope for here anyways (and maybe in the category of "don't fix it if it's not broken"?)